### PR TITLE
KAFKA-4578: Upgrade notes for 0.10.2.0

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @ewencp. @guozhangwang @miguno, can you please check that the Streams upgrade bit is in the right place now that we have proper upgrade notes? I am not sure what was the intended location and style.

File: docs/upgrade.html [hunk 1]
```diff
@@ -14,31 +14,73 @@
...
+
+<p><b>Note:</b> If you are willing to accept downtime, you can simply take all the brokers down, update the code and start all of them. They will start with the new protocol by default.
+
+<p><b>Note:</b> Bumping the protocol version and restarting can be done any time after the brokers were upgraded. It does not have to be immediately after.
+
+<ol>
+    <li>Upgrading a Kafka Streams Applications:
+        <ul>
+            <li>You need to recompile your code. Just swapping the jar file will not work and will break your appliation.</li>
+            <li>If you use a custom timestamp extractor, you will need to update this code, because the <code>TimestampExtractor</code> interface got changed.</li>
```
Comment (ewencp) [inline docs/upgrade.html]: nit: i realize this was just moved, but "interface got changed" should be either "interface was changed" or just "interface changed".

File: docs/upgrade.html [hunk 2]
```diff
@@ -14,31 +14,73 @@
...
+        should not be set in the streams app any more. If the Kafka cluster is secured, Streams apps must have the required security privileges to create new topics.</li>
+    <li>Several new fields including "security.protocol", "connections.max.idle.ms", "retry.backoff.ms", "reconnect.backoff.ms" and "request.timeout.ms" were added to
+        StreamsConfig class. User should pay attenntion to the default values and set these if needed. For more details please refer to <a id="streamsconfigs" href="#streamsconfigs">3.5 Kafka Streams Configs</a>.</li>
+</ul>
+
+<h5><a id="upgrade_1020_new_protocols" href="#upgrade_1020_new_protocols">New Protocol Versions</a></h5>
+<ul>
+    <li> KIP-88: OffsetFetchRequest v2 supports retrieval of offsets for all topics if the <code>topics</code> array is set to <code>null</code>. </li>
+    <li> KIP-88: OffsetFetchResponse v2 introduces a top-level <code>error_code</code> field. </li>
+    <li> KIP-103: UpdateMetadataRequest v2 introduces a <code>listener_name</code> field to the elements of the <code>end_points</code> array. </li>
```
Comment (ewencp) [inline docs/upgrade.html]: UpdateMetadataRequest hit v3, not v2. The KIP had the incorrect version so I've fixed in there as well.

File: docs/upgrade.html [hunk 3]
```diff
@@ -14,31 +14,73 @@
...
+    <li>The Zookeeper dependency was removed from the Streams API. The Streams API now uses the Kafka protocol to manage internal topics instead of
+        modifying Zookeeper directly. This eliminates the need for privileges to access Zookeeper directly and "StreamsConfig.ZOOKEEPER_CONFIG"
+        should not be set in the streams app any more. If the Kafka cluster is secured, Streams apps must have the required security privileges to create new topics.</li>
+    <li>Several new fields including "security.protocol", "connections.max.idle.ms", "retry.backoff.ms", "reconnect.backoff.ms" and "request.timeout.ms" were added to
+        StreamsConfig class. User should pay attenntion to the default values and set these if needed. For more details please refer to <a id="streamsconfigs" href="#streamsconfigs">3.5 Kafka Streams Configs</a>.</li>
+</ul>
+
+<h5><a id="upgrade_1020_new_protocols" href="#upgrade_1020_new_protocols">New Protocol Versions</a></h5>
+<ul>
+    <li> KIP-88: OffsetFetchRequest v2 supports retrieval of offsets for all topics if the <code>topics</code> array is set to <code>null</code>. </li>
```
Comment (ewencp) [inline docs/upgrade.html]: nit: you could link to the relevant KIPs.

**Review by ewencp — APPROVED** [review]
@ijuma A few minor nits, but LGTM

**Comment by ijuma** [comment]
Thanks for the review @ewencp, I addressed your feedback.

**Review by ewencp — APPROVED** [review]
LGTM. Are we just waiting on a streams reviewer to take a pass?

File: docs/upgrade.html [hunk 4]
```diff
@@ -14,31 +14,73 @@
...
-    <li>Several new fields including "security.protocol", "connections.max.idle.ms", "retry.backoff.ms", "reconnect.backoff.ms" and "request.timeout.ms" were added to StreamsConfig class. User should pay attenntion to the default values and set these if needed. For more details please refer to <a id="streamsconfigs" href="#streamsconfigs">3.5 Kafka Streams Configs</a>.</li>
+    <li>The Zookeeper dependency was removed from the Streams API. The Streams API now uses the Kafka protocol to manage internal topics instead of
+        modifying Zookeeper directly. This eliminates the need for privileges to access Zookeeper directly and "StreamsConfig.ZOOKEEPER_CONFIG"
+        should not be set in the streams app any more. If the Kafka cluster is secured, Streams apps must have the required security privileges to create new topics.</li>
```
Comment (miguno) [inline docs/upgrade.html]: typo: "streams app" -> "Streams app" (was already misspelled in the original text)

**Comment by miguno** [comment]
LGTM, though perhaps @guozhangwang can take a quick look, too.

**Comment by ijuma** [comment]
Thanks for the reviews, I merged this. Someone from Streams can submit a follow-up if additional changes are required. Thinking about it, it would be a good idea to mention that Streams only supports brokers with version 0.10.1 and above (unlike other clients).

**Pull request closed without merging**